website/versioned_docs/version-5.x/layout-overview.md

```diff
@@ -0,0 +1,95 @@
+---
+id: layout-overview
+title: Layout Engine
+sidebar_label: Overview
+---
+
+## Introduction
+
+[Github repository](https://github.com/single-spa/single-spa-layout/)
+
+The `single-spa-layout` npm package is an optional add-on to single-spa. The layout engine provides a routing API that controls your top level routes, applications, and dom elements. Using single-spa-layout makes it easier to accomplish the following:
+
+- DOM placement and ordering of applications.
+- Loading UIs when applications are downloaded.
+- Default routes for Not Found / 404 pages.
+- Transitions between routes (implementation pending).
+
+The layout engine performs two major tasks:
+
+1. Generate [single-spa registration config](./api#configuration-object) from an HTML Element and/or JSON object.
+1. Listen to [routing events](./api#events) to ensure that all DOM elements are laid out correctly before the single-spa applications are mounted.
+
+`single-spa-layout` is 3.2kb gzipped (9kb ungzipped).
+
+## Installation
+
+You only need to install the layout engine into your [root config](./configuration) (not in any other application).
+
+```sh
+npm install --save single-spa-layout@beta
+
+yarn add single-spa-layout@beta
+```
+
+### Project status
+
+`single-spa-layout` is new and currently released under the `beta` dist-tag on npm. We are gathering feedback and improving the layout engine in preparation for a stable release. Although we do not expect the layout engine to change drastically, we do not recommend using it in a production setting until it is released as stable.
+
+### Browser support
+
+`single-spa-layout` works in all browsers supported by single-spa, including IE11.
+
+### Requirements
+
+You must use single-spa@>=5.4.0 in order for the layout engine to work. Additionally, you may not provide custom `domElementGetter` functions for any of your single-spa applications, as those override the configuration within single-spa-layout.
```

Comment:
Oh interesting. This would make it tricky to migrate to if an org had relied on domElementGetters for all their applications, right? One would have to find a way to remove domElementGetter, deploy updated application assets to prod but not update the prod importmap, and can then publish the updated root config and update all the imports at that time. I'm not good at visualizing this all conceptually, so please let me know if I missed something obvious. 😅

Reply:
Yes if an organization is heavily using domElementGetters they will have to migrate. Here's what a migration could look like:

Before:
```js
function domElementGetter() {
  let navEl = document.getElementById('our-nav')
  if (!navEl) {
    const navEl = document.createElement('nav');
    navEl.id = 'our-nav';
    navEl.style.position = 'fixed';
    document.body.appendChild(navEl)
  }

  return navEl
}
```

Migration / Transition period:
```js
function domElementGetter() {
  let navEl = document.getElementById('single-spa-application:our-nav') || document.getElementById('our-nav')
  if (!navEl) {
    const navEl = document.createElement('nav');
    navEl.id = 'our-nav';
    navEl.style.position = 'fixed';
    document.body.appendChild(navEl)
  }

  return navEl
}
```

After:

(Delete the domElementGetter entirely)